python/tweets/searcher.py

```diff
@@ -0,0 +1,97 @@
+# This is a "tweet grabber" script - when run, it searches the Twitter archive and
+# finds all tweets matching a certain query, and then stores them in a mongoDB mLab database.
+
+
+# Import modules (you may need to `pip install` these at the command line):
+# - tweepy, for interacting with the Twitter API
+# - json, for working with the tweets themselves (they are returned as json objects)
+# - pymongo, for connecting to a mongoDB mLab database where we'll store the tweets
+import tweepy, json, pymongo
+
+
+# Set up API - replace keys with your own: https://apps.twitter.com/
+CONSUMER_KEY = "<CONSUMER_KEY>"
+CONSUMER_SECRET = "<CONSUMER_SECRET>"
+ACCESS_KEY = "<ACCESS_KEY>"
+ACCESS_SECRET = "<ACCESS_TOKEN>"
+auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
+auth.set_access_token(ACCESS_KEY, ACCESS_SECRET)
+api = tweepy.API(auth)
+
+
+# Bath, England place id for twitter api is "1db4f0a70fc5c9db", found using api.geo_search:
+# api.geo_search(query="Bath, England", granularity="city")[0]
+# We'll search for tweets from here
+search_query = "place:1db4f0a70fc5c9db"
+
+# Max tweets to grab in one go (just a big number)
+max_tweets = 1000000000
+# Set to max per query that API allows (100)
+tweets_per_query = 100
+
+# Connect to the mLab database
+client = pymongo.MongoClient("mongodb://shareduser:shareduser@ds123614.mlab.com:23614/bmlm-parking-tweets")
+db = client["bmlm-parking-tweets"]
+
+# If we already have some tweets, get the date of the most recent, and only look for tweets after that
+try:
+    most_recent = [tweet for tweet in db.tweets.find().sort("id", pymongo.DESCENDING)]
```

Comment:
You should be able to do `db.tweets.find_one(sort=[("id", pymongo.DESCENDING)])` and not have to load the entire collection in memory. Writing this blindly without testing, so you may have to tweak the command a bit.

Do we have both "id", from the API, and "_id" from Mongo in these documents then?

Reply:
Your blind code is spot on!

We do have both `id` (the tweet id) and `_id` (the mongo document id). We don't necessarily need both. But we have both. We also have a whole load of other unhelpful metadata in each document though - I want to discuss with the group at some point what we should/shouldn't keep.

Reply:
We'd probably want to keep both, but maybe rename the `id` one at a later point in time to spare us some confusion?

As for the metadata, when pulling data out of mongo we can do what's known as projections to specify exactly what data we want from each document for when we're actually using the data. So I don't imagine this will be too difficult unless there's a very low mongo disk space limit.

Reply:
I have a disk limit of 496MB. Currently at about 101MB after 3 weeks or so (although some of that turns out to be duplicates). When we get near the limit we can either pull it all out and store locally somewhere; or another option is reduce size of each document by storing less metadata.